src/xunit.performance.api/XunitRunner.cs

```diff
@@ -4,108 +4,95 @@
+                throw new ArgumentException("The test type name cannot be white space.");
+
             // Create a runner for the specifid assembly.
-            using(AssemblyRunner runner = AssemblyRunner.WithoutAppDomain(assemblyPath))
+            using (var runner = AssemblyRunner.WithoutAppDomain(assemblyPath))
             {
-                // Setup callbacks.
-                runner.OnDiscoveryComplete = OnDiscoveryComplete;
-                runner.OnExecutionComplete = OnExecutionComplete;
-                runner.OnTestStarting = OnTestStarting;
-                runner.OnTestFailed = OnTestFailed;
-                runner.OnTestSkipped = OnTestSkipped;
+                // Start out assuming success; we'll set this to 1 if we get a failed test
+                var result = new int[] { 0 };
 
-                Console.WriteLine("\nDiscovering performance tests in {0}.", assemblyPath);
+                // Use an event to know when we're done
+                using (var finished = new ManualResetEvent(false))
+                {
+                    // We use consoleLock because messages can arrive in parallel, so we want to make sure we get
+                    // consistent console output.
+                    var consoleLock = new object();
 
-                // Start running tests.
-                runner.Start(null);
+                    SetupRunnerCallbacks(runner, finished, consoleLock, result);
+                    WriteInfoLine($"Discovering performance tests in \"{assemblyPath}\".");
 
-                // Wait for tests to complete.
-                _finished.WaitOne();
-                _finished.Dispose();
+                    runner.Start(typeName);
 
-                // Wait for the assembly runner to go idle.
-                ManualResetEvent waitHandle = new ManualResetEvent(false);
-                while(runner.Status != AssemblyRunnerStatus.Idle)
-                {
-                    waitHandle.WaitOne(10);
+                    // Wait for tests to complete.
+                    finished.WaitOne();
                 }
 
-                // Return overall status.
-                return _result;
-            }
-        }
+                // Wait for the assembly runner to go idle.
+                using (var waitHandle = new ManualResetEvent(false))
+                    while (runner.Status != AssemblyRunnerStatus.Idle)
+                        waitHandle.WaitOne(10);
 
-        private void OnDiscoveryComplete(DiscoveryCompleteInfo info)
-        {
-            lock (_consoleLock)
-            {
-                Console.WriteLine($"Running {info.TestCasesToRun} of {info.TestCasesDiscovered} tests...");
+                // Return overall status.
+                return result[0];
             }
         }
 
-        private void OnExecutionComplete(ExecutionCompleteInfo info)
+        private static void SetupRunnerCallbacks(AssemblyRunner runner, ManualResetEvent manualResetEvent, object consoleLock, int[] result)
```

Comment:
I believe that these arguments were originally instance variables, which would make this much cleaner.